src/Core/Config/CoreConfigCreator.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+namespace SilverStripe\Core\Config;
+
+use SilverStripe\Config\Collections\CachedConfigCollection;
+use SilverStripe\Config\Collections\MemoryConfigCollection;
+use SilverStripe\Config\Transformer\PrivateStaticTransformer;
+use SilverStripe\Config\Transformer\YamlTransformer;
+use SilverStripe\Control\Director;
+use SilverStripe\Core\Config\Middleware\ExtensionMiddleware;
+use SilverStripe\Core\Config\Middleware\InheritanceMiddleware;
+use SilverStripe\Core\Manifest\ClassLoader;
+use SilverStripe\Core\Manifest\ModuleLoader;
+use Symfony\Component\Cache\Adapter\FilesystemAdapter;
+use Symfony\Component\Finder\Finder;
+
+/**
+ * Factory for silverstripe configs
+ */
+class CoreConfigCreator
+{
+    /**
+     * @var static
+     */
+    protected static $inst = null;
+
+    /**
+     * @return static
+     */
+    public static function inst()
+    {
+        if (!self::$inst) {
+            self::$inst = new static();
+        }
+        return self::$inst;
+    }
+
+    /**
+     * Create root application config.
+     * This will be an immutable cached config,
+     * which conditionally generates a nested "core" config.
+     *
+     * @param bool $flush
+     * @return CachedConfigCollection
+     */
+    public function createRoot($flush)
+    {
+        $instance = new CachedConfigCollection();
+
+        // Set root cache
+        $instance->setPool(new FilesystemAdapter('configcache', 0, getTempFolder()));
```

Comment:
Is there a plan for how to make this more pluggable? I guess it'll be easier once we create an `App` object as part of `main.php` bootstrap, where you could manually set services before relying on DI?